software/apps/systolic/matmul/main.c

```diff
@@ -0,0 +1,186 @@
+// Copyright 2022 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// Author: Gua Hao Khov, ETH Zurich
+
+#include <stdint.h>
+#include <string.h>
+
+#include "alloc.h"
+#include "encoding.h"
+#include "printf.h"
+#include "runtime.h"
+#include "synchronization.h"
+#include "systolic/matmul.h"
+
+// Dimensions of matrices
+#define DIM_M 32
+#define DIM_N 32
+#define DIM_P 32
+
+uint32_t *grid_mapping;
+
+int32_t *matrix_A;
+int32_t *matrix_B;
+
+uint32_t rep_count;
+
+systolic_matrix_t *syst_matrix_A;
+systolic_matrix_t *syst_matrix_B;
+systolic_matrix_t *syst_matrix_C;
+
+void generate_gradient_matrix(int32_t **matrix, uint32_t num_rows,
+                              uint32_t num_cols) {
+  int32_t *new_matrix = (int32_t *)simple_malloc(num_rows * num_cols * 4);
+  for (uint32_t y = 0; y < num_rows; ++y) {
+    for (uint32_t x = 0; x < num_cols; ++x) {
+      new_matrix[y * num_cols + x] = (int32_t)(y + x);
+    }
+  }
+  *matrix = new_matrix;
+}
+
+void print_matrix(int32_t const *matrix, uint32_t num_rows,
+                  uint32_t num_columns) {
+  printf("Matrix at 0x%8X\n", (uint32_t)matrix);
+  for (uint32_t i = 0; i < num_rows; ++i) {
+    for (uint32_t j = 0; j < num_columns; ++j) {
+      printf("%5d ", matrix[i * num_columns + j]);
+    }
+    printf("\n");
+  }
+}
+
+int main() {
+  uint32_t core_id = mempool_get_core_id();
+  uint32_t num_cores = mempool_get_core_count();
+  uint32_t tile_id = core_id / 4;
+
+  // Initialize synchronization variables
+  mempool_barrier_init(core_id);
+
+  // Initialization
+  mempool_init(core_id, num_cores);
+
+  // Allocate systolic grid mapping
+  if (core_id == 0) {
+    grid_mapping = (uint32_t *)simple_malloc(num_cores * 4);
+    if (num_cores != SYSTOLIC_SIZE * SYSTOLIC_SIZE) {
+      printf("SYSTOLIC_SIZE does not match core count!\n");
+      return -1;
+    }
+  }
+
+  // Assign grid position (row wise)
+  // uint32_t col_idx = core_id % SYSTOLIC_SIZE;
+  // uint32_t row_idx = core_id / SYSTOLIC_SIZE;
+
+  // Assign grid position (col wise)
+  uint32_t col_idx = core_id / SYSTOLIC_SIZE;
+  uint32_t row_idx = core_id % SYSTOLIC_SIZE;
+
+  // 16 CORES only
+  // Assign grid position (tile wise)
+  // uint32_t col_idx;
+  // uint32_t row_idx;
+  // if (core_id < 4) {
+  //   col_idx = core_id % 2;
+  //   row_idx = core_id / 2;
+  // } else if (core_id < 8) {
+  //   col_idx = core_id % 2 + 2;
+  //   row_idx = core_id / 6;
+  // } else if (core_id < 12) {
+  //   col_idx = core_id % 2;
+  //   row_idx = core_id / 10 + 2;
+  // } else {
+  //   col_idx = core_id % 2 + 2;
+  //   row_idx = core_id / 14 + 2;
+  // }
```

Comment:
What is this code? Is this used? If it is specific to a deprecated `minpool-systolic` configuration, we could add an `ifdef` in here. Otherwise, I propose we delete this and left it in the git history.